coaster/sqlalchemy.py

```diff
@@ -166,6 +166,26 @@ def __init__(self, *args, **kw):
         if not self.name:
             self.make_name()
 
+    @classmethod
+    def get(cls, name):
+        """Get an instance matching the name"""
+        return cls.query.filter_by(name=name).one_or_none()
+
+    @classmethod
+    def upsert(cls, name, **fields):
+        """Insert or update an instance"""
+        instance = cls.get(name)
+        if instance:
+            for f in fields:
+                if hasattr(instance, f):
+                    setattr(instance, f, fields[f])
+                else:
+                    raise TypeError("'{arg}' is an invalid keyword argument for {instance_type}".format(arg=f, instance_type=instance.__class__.__name__))
```

Comment:
Good idea. Since it's an internal helper method not exposed to downstream subclasses, it should be underscore prefixed, so maybe `_set_fields`. You may also want to investigate how SQLAlchemy records dirty fields: (a) when the value changes, or (b) only if the value changes to a new value? If it doesn't do (b), you should do that for better db write efficiency.